authz/audit/stdout/stdout_logger.go

```diff
@@ -0,0 +1,96 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package stdout defines an stdout audit logger.
+package stdout
+
+import (
+	"encoding/json"
+	"log"
+	"time"
+
+	"google.golang.org/grpc/authz/audit"
+	"google.golang.org/grpc/grpclog"
+)
+
+var grpcLogger = grpclog.Component("authz-audit")
+
+func init() {
+	audit.RegisterLoggerBuilder(&loggerBuilder{})
+}
+
+type event struct {
+	FullMethodName string `json:"fullMethodName"`
+	Principal      string `json:"principal"`
+	PolicyName     string `json:"policyName"`
+	MatchedRule    string `json:"matchedRule"`
+	Authorized     bool   `json:"authorized"`
+	Timestamp      string `json:"timestamp"` // Time when the audit event is logged via Log method
+}
+
+// logger implements the audit.Logger interface by logging to standard output.
+type logger struct {
+}
+
+// Log marshals the audit.Event to json and prints it to standard output.
+func (logger *logger) Log(event *audit.Event) {
+	jsonBytes, err := json.Marshal(convertEvent(event))
```

Comment:
Admittedly, it's not reflected in the current revision of the gRFC. But I believe we've discussed that we want this json to be wrapped by an outer json object, i.e. {"grpc_audit_logging": <jsonBytes here>}.

Reply:
changing to `grpc_audit_log` as it's the current version from gRFC